cmd/cloud-controller-manager/nodeipamcontroller.go

```diff
@@ -75,6 +75,11 @@ func startNodeIpamController(initContext app.ControllerInitContext, ccmConfig *c
 		return nil, false, nil
 	}
 
+	// Cannot run cloud ipam controller if cloud provider is nil (--cloud-provider not set or set to 'external')
+	if cloud == nil && ccmConfig.ComponentConfig.KubeCloudShared.CIDRAllocatorType == string(ipam.CloudAllocatorType) {
+		return nil, false, errors.New("--cidr-allocator-type is set to 'CloudAllocator' but cloud provider is not configured")
```

Comment:
can CIDRAllocatorType be explicitly set to an empty string or `None` or `External` or something that indicates the caller knows/intends for this process to not handle allocating these? if so, should we include those details in the error? same question in k-c-m

Reply:
> can CIDRAllocatorType be explicitly set to an empty string or None or External or something that indicates the caller knows/intends for this process to not handle allocating these?

I don't think so -- the current valid values are: `RangeAllocator` (default), `MultiCIDRRangeAllocator`, `CloudAllocator`, `IPAMFromCluster` (legacy) and `IPAMFromCloud` (legacy). You can opt-out of this controller entirely, by unsetting `--allocate-node-cidrs` though, are you suggesting we include a note here about using `--allocate-node-cidrs` to opt out of this controller entirely?

Reply:
> You can opt-out of this controller entirely, by unsetting --allocate-node-cidrs though

ah, ok... so this error is only presented to people invoking the binary and explicitly setting `--allocate-node-cidrs` AND setting `--cidr-allocator-type=CloudAllocator`, both of which are non-default settings?

Reply:
yep 

https://github.com/kubernetes/kubernetes/blob/75e39073a3ecd975eabd8bdb4f1fff84ebad1dce/cmd/controller-manager/app/options/kubecloudshared.go#L57-L58


and not specificying the GCE config